### ISSUE TITLE
Automaticaly adding modifiers in keymap

### ISSUE THREAD

**Issue by danyspin97** [k0kubun/xremap#102]
I am replacing `rightalt + n` with `left` key and a couple more. However, it requires a new remap for every added modifier combination.

```
keymap:
  - name: Arrows
    remap:
      alt_r-n: left
      alt_r-e: down
      alt_r-i: up
      alt_r-o: right
      super-alt_r-n: super-left
      super-alt_r-e: super-down
      super-alt_r-i: super-up
      super-alt_r-o: super-right
      super-shift_l-alt_r-n: super-shift_l-left
      super-shift_l-alt_r-e: super-shift_l-down
      super-shift_l-alt_r-i: super-shift_l-up
      super-shift_l-alt_r-o: super-shift_l-right
```

Is there an way to just add the first four remaps and make them work with any other modifier automatically? I have tried with `modmap` but I couldn't make it work.

**Comment by k0kubun** [comment]
Yeah, I think `alt_r-n: left` alone should remap `whatever-alt_r-n` to `whatever-left`. It's currently not working that way due to a premature optimization in the implementation.

I'll work on this when I can allocate time to work on this. Meanwhile, please use the current workaround and also consider generating it with a template as needed.

**Comment by k0kubun** [comment]
Update: As I'm currently focusing on a big change for another OSS, I'm planning to work on this issue on Aug 13th. If I receive a pull request before that, I can take a look at that too.

**Comment by k0kubun** [comment]
Released https://github.com/k0kubun/xremap/pull/145 in v0.6.0. Enjoy.

**Comment by nullbyto** [comment]
For report: i tried it with release v0.7.0 and downloaded v0.6.0, it's not working for me. I'm not sure if I have to do something extra.

**Comment by k0kubun** [comment]
In addition to the version, could you report the following information?

* Platform: X11, GNOME Wayland, or Sway
* Your config (preferably minimum one)
* Output of `RUST_LOG=debug xremap`

For example, using xremap v0.7.0 on X11, my config to test this behavior is:

```yml
keymap:
  - remap:
      alt_r-n: left
```

and the `RUST_LOG=debug xremap` is:

```
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] => 1: KEY_LEFTCTRL
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 1: KEY_LEFTCTRL
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] => 1: KEY_RIGHTALT
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 1: KEY_RIGHTALT
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] => 1: KEY_N
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 0: KEY_LEFTCTRL
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 0: KEY_RIGHTALT
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 1: KEY_LEFT
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 0: KEY_LEFT
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 1: KEY_LEFTCTRL
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 1: KEY_RIGHTALT
^[[D[2022-08-14T22:23:35Z DEBUG xremap::event_handler] => 0: KEY_LEFTCTRL
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 0: KEY_LEFTCTRL
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] => 0: KEY_N
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 0: KEY_N
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] => 0: KEY_RIGHTALT
[2022-08-14T22:23:35Z DEBUG xremap::event_handler] 0: KEY_RIGHTALT
```

As you can see, while `Ctrl-Alt-N` is pressed (ctrl is not needed), `Left` is emitted successfully.

**Comment by k0kubun** [comment]
Well, I guess you want to leave the extra modifier, e.g. `Ctrl-Alt-N` should be `Ctrl-Left`? I'm not sure if it's what you mean, but I can work on fixing this.

**Comment by nullbyto** [comment]
Yes thats what i mean. So no need to write all the combinations. It makes configurating cleaner.

**Comment by k0kubun** [comment]
Addressed it in v0.7.1. Please try that version.

**Comment by nullbyto** [comment]
Works perfectly! Thanks for your great work.
